### PR TITLE
[Rest Api Compatibility] Allow transforming warnings per test

### PR DESCRIPTION
**Pull request by pgomulka**
Warning related transformations missed the possibility to apply per single test only.
Also a warning changed in #67158 for indices.close so this PR also applies the transformation for 7.x test

relates #51816

### CODE REVIEW

**Comment by pgomulka** [comment]
@elasticmachine run elasticsearch-ci/part-1

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/rest/compat/RestCompatTestTransformTask.java [hunk 1]
```diff
@@ -298,6 +307,14 @@ public void addAllowedWarningRegex(String... allowedWarningsRegex) {
         transformations.add(new InjectAllowedWarnings(true, Arrays.asList(allowedWarningsRegex)));
     }
 
+    /**
+     * Adds one or more allowed regular expression warnings
+     * @param allowedWarningsRegex the regex warning(s) to add
```
Comment (jakelandis) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/rest/compat/RestCompatTestTransformTask.java:312]: nit: javadoc for testName

**Review by jakelandis — APPROVED** [review]
LGTM

**Comment by pgomulka** [comment]
@elasticmachine update branch

**Pull request merged**